graphjet-core/src/main/java/com/twitter/graphjet/algorithms/filters/TweetAuthorFilter.java

```diff
@@ -0,0 +1,121 @@
+/**
+ * Copyright 2018 Twitter. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.twitter.graphjet.algorithms.filters;
+
+import com.twitter.graphjet.algorithms.RecommendationRequest;
+import com.twitter.graphjet.bipartite.LeftIndexedMultiSegmentBipartiteGraph;
+import com.twitter.graphjet.bipartite.api.EdgeIterator;
+import com.twitter.graphjet.hashing.SmallArrayBasedLongToDoubleMap;
+import com.twitter.graphjet.stats.Counter;
+import com.twitter.graphjet.stats.StatsReceiver;
+
+import it.unimi.dsi.fastutil.longs.LongOpenHashSet;
+import it.unimi.dsi.fastutil.longs.LongSet;
+
+/**
+ * This filter filters tweets by tweet authors, depending on whether the node author is
+ * on the blacklist or the whitelist.
+ *
+ * - For whitelist authors, only tweets from the whitelisted authors will pass the filter.
+ *   However, if the whitelist is empty, the filter will pass all tweets.
+ * - For blacklist authors, tweets authored by blacklisted authors will be filtered.
+ */
+public class TweetAuthorFilter extends ResultFilter {
+
+  private boolean isWhitelistAllTweets;
+  private LongSet whitelistedTweets;
+  private LongSet blacklistedTweets;
+
+  private Counter blacklistFilterCount = scopedStatsReceiver.counter("blacklist_filtered");
+  private Counter whitelistFilterCount = scopedStatsReceiver.counter("whitelist_filtered");
+
+  public TweetAuthorFilter(
+      LeftIndexedMultiSegmentBipartiteGraph leftIndexedBipartiteGraph,
+      LongSet whitelistTweetAuthors,
+      LongSet blacklistTweetAuthors,
+      StatsReceiver statsReceiver) {
+    super(statsReceiver);
+    this.isWhitelistAllTweets = whitelistTweetAuthors.isEmpty();
+    LongSet dedupedWhitelistAuthors = dedupWhitelistAuthors(whitelistTweetAuthors, blacklistTweetAuthors);
+    this.whitelistedTweets = getTweetsByAuthors(leftIndexedBipartiteGraph, dedupedWhitelistAuthors);
+    this.blacklistedTweets = getTweetsByAuthors(leftIndexedBipartiteGraph, blacklistTweetAuthors);
```

Comment:
let us query for blacklist tweets only when whitelist is empty. Otherwise, we are querying for blacklist tweets all the time.